src/modules/ce/components/ReferralStep.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useMemo } from 'react';
+import { useNavigate } from 'react-router-dom';
+import CommonCard from '@/components/elements/CommonCard';
+import Loading from '@/components/elements/Loading';
+import NotFound from '@/components/pages/NotFound';
+import useSafeParams from '@/hooks/useSafeParams';
+import ReferralStepAssignee from '@/modules/ce/components/ReferralStepAssignee';
+import { emptyErrorState } from '@/modules/errors/util';
+import DynamicForm from '@/modules/form/components/DynamicForm';
+import DynamicView from '@/modules/form/components/viewable/DynamicView';
+import {
+  createInitialValuesFromSavedValues,
+  getItemMap,
+} from '@/modules/form/util/formUtil';
+import { ProjectDashboardRoutes } from '@/routes/routes';
+import {
+  CeReferralStatus,
+  CeReferralStepStatus,
+  useGetCeReferralStepQuery,
+  useSubmitCeReferralStepMutation,
+} from '@/types/gqlTypes';
+import { generateSafePath } from '@/utils/pathEncoding';
+
+interface Props {}
+const ReferralStep: React.FC<Props> = ({}) => {
+  const { projectId, opportunityId, referralId, stepId } = useSafeParams() as {
+    projectId: string;
+    opportunityId: string;
+    referralId: string;
+    stepId: string;
+  };
+  const navigate = useNavigate();
+
+  const {
+    data: { ceReferralStep: step } = {},
+    loading: fetchLoading,
+    error: fetchError,
+  } = useGetCeReferralStepQuery({
+    variables: {
+      id: stepId,
+    },
+  });
+
+  const [submit, { loading: submitLoading, error: submitError }] =
+    useSubmitCeReferralStepMutation({
+      onCompleted: (data) => {
+        const status = data.submitCeReferralStep?.referral.status;
+        const wayfind =
+          status &&
+          [CeReferralStatus.Rejected, CeReferralStatus.Accepted].includes(
+            status
+          );
+
+        navigate({
+          pathname: generateSafePath(ProjectDashboardRoutes.REFERRAL_STEPS, {
+            projectId,
+            opportunityId,
+            referralId,
+          }),
+          search: wayfind
+            ? new URLSearchParams({ wayfinding: 'true' }).toString()
+            : undefined,
+        });
+      },
+    });
+
+  const { name, status, formDefinition, submittedValues } = step || {};
+
+  const itemMap = useMemo(
+    () => formDefinition && getItemMap(formDefinition.definition),
+    [formDefinition]
+  );
+
+  const formState =
+    itemMap &&
+    submittedValues &&
+    createInitialValuesFromSavedValues(itemMap, submittedValues);
+
+  if (fetchLoading) return <Loading />;
+  if (fetchError) throw fetchError;
+  if (submitError) throw submitError;
+  if (!step || !formDefinition) return <NotFound />;
+
+  return (
+    <CommonCard title={name}>
+      <Stack gap={2}>
+        <ReferralStepAssignee step={step} />
+        <Divider />
+        {status === CeReferralStepStatus.InProgress && (
+          <DynamicForm
+            definition={formDefinition.definition}
+            onSubmit={(event) => {
```

Comment:
Nit renaming "event", since its not an event object it is a custom `DynamicFormSubmitInput` record